corehq/apps/locations/expressions.py

```diff
@@ -0,0 +1,28 @@
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.userreports.specs import TypeProperty
+from corehq.util.quickcache import quickcache
+from dimagi.ext.jsonobject import JsonObject
+
+
+class LocationTypeSpec(JsonObject):
+    type = TypeProperty('location_type_name')
+
+    def __call__(self, item, context=None):
+        doc_id = item.get('_id', None)
```

Comment:
I was thinking about this more - I think that we shouldn't hard code `_id` here, but make the id lookup an expression similar to `related_doc`. That will make it much more useful since it will be able to be used on anything with a `location_id` as opposed to just location documents